pkg/vsphere/disk/disk_manager.go

```diff
@@ -223,6 +223,20 @@ func (m *Manager) Create(ctx context.Context, newDiskURI string,
 //	return nil
 // }
 
+func (m *Manager) Attach(ctx context.Context, spec *types.VirtualDisk) error {
+	machineSpec := configureDeviceSpec(ctx, *m.vm, types.VirtualDeviceConfigSpecOperationAdd, types.VirtualDeviceConfigSpecFileOperationCreate, spec)
+
+	_, err := tasks.WaitAndRetryForResult(ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
```

Comment:
Also, this happens at the wrong layer. But, lets chat about that in a different PR which is solely integration of this helper.